blueprints/controllers/ikea_e1812/ikea_e1812.yaml

```
@@ -59,197 +70,279 @@ blueprint:
       default: []
       selector:
         action:
-    # inputs for looping custom actions on long button press events until the corresponding release event is received
     button_long_loop:
       name: Long button press - loop until release
       description: Loop the action until the button is released.
       default: false
       selector:
         boolean:
-    # inputs for enabling double press events
+    button_long_max_loops:
+      name: Long Button press - Maximum loops
+      description: Maximum amount of repeats.
+      default: 500
+      selector:
+        number:
+          min: 0
+          max: 5000
+          mode: slider
+          step: 1
     button_double_press:
       name: Expose button double press event
-      description: Choose whether or not to expose the virtual double press event for the button. Turn this on if you are providing an action for the button double press event.
+      description: Choose whether or not to expose the virtual double press event
+        for the button. Turn this on if you are providing an action for the button
+        double press event.
+      default: false
+      selector:
+        boolean:
+    button_double_initial_press:
+      name: Run short press action at once.
+      description: Choose whether or not to run the short press action at once, without waiting for the 2nd press.
+        When a second press is received the double press action will be run at once and the short press action will stop if it's still running.
       default: false
       selector:
         boolean:
-    # helpers used to properly recognize the remote button events
+    #Only requiered since it seems impossible to get the automations attribute dynamicly via a blueprint. 
     helper_last_controller_event:
-      name: (Double press action Required) Helper - Last Controller Event
-      description: Input Text used to store the last event fired by the controller. See docs for more.
+      name: (Requieded to counter double triggers) Helper - Last Controller Event
+      description: Input Text used to store the last event fired by the controller.
+        See docs for more.
       default: ''
       selector:
         entity:
-          domain: input_text
-    helper_double_press_delay:
+          domain: input_text            
+    double_press_delay:
```

```suggestion
          domain: input_text
    helper_double_press_delay:
```

keep the same format. Moreover, I'd prefer not to rename current inputs, if not strictly required, since it requires users which already are using this blueprint to reconfigure their automations from scratch, since the YAML config won't be valid anymore

---

A helper is a input_something in home assistant, the delay is a blueprint variable and thus not the same thing as a helper. I found this confusing :)

---

you are right in that it would break stuff for current users. So let's keep it like that.